src/Service/Services/MetadataProviders/CosmosSqlMetadataProvider.cs

```diff
@@ -310,20 +307,40 @@ public bool TryGetEntityNameFromPath(string entityPathName, [NotNullWhen(true)]
         /// <inheritdoc />
         public string GetEntityName(string graphQLType)
         {
-            if (_entities.ContainsKey(graphQLType))
+            if (_runtimeConfig.Entities.ContainsKey(graphQLType))
             {
                 return graphQLType;
             }
 
-            if (!_graphQLSingularTypeToEntityNameMap.TryGetValue(graphQLType, out string? entityName))
+            // Cosmos allows you to have a different GraphQL type name than the entity name in the config
+            // and we use the `model` directive to map between the two. So if the name originally provided
+            // doesn't match any entity name, we try to find the entity name by looking at the GraphQL type
```

Comment:
What happens when model directive name doesn't match the singular type name specified in the config?

Reply:
What if GraphQL Schema type is `graphqlTypeA`, model name is `modelEntityA`, entity name is `EntityA` and singular property is `SingularEntityA`? => essentially, nothing matches?
What scenario does the test for `StarAlias` entity resemble?

Reply:
For `StarAlias` we have a GraphQL type defined as so:

```graphql
type Star @model(name:"StarAlias")
```

When a request comes in from a GraphQL client it will do a query (or mutation) against the GraphQL type of `Star`. This is what is received when we get into the parsing phase to build the Cosmos query and then it'll attempt to look that up in the config as an entity, which fails as the entity is `StarAlias`, so instead we fallback to looking at the `ObjectType` definitions in the GraphQL schema, find the `model` types and then find the one with the GraphQL type name of `Star` to then extract the entity name from the `model` directive and find it in the `RuntimeConfig`.

The only edge case I can think of for this is if you created a GraphQL type of `type Foo @model("Bar")`, and then in the JSON you define the singular of `Bar` to be `Baz` then the query/mutation naming should be exposed in the generated schema as `baz`, but the GraphQL type that is to be returned from `baz_by_pk` is probably still `Foo` (since with Cosmos we only generate the query/mutation types not the ObjectTypes as they are in the provided schema), resulting in `baz_by_id(...): Foo`, so there's probably all kinds of weirdness in the schema at that point.

So that _may_ be a regression as a result of this change, but given that the singular name isn't _truly_ respected, meaning the GraphQL type is still `Foo`, I'd have thought we'd see a bug by name as `baz_by_id(...): Foo` would indicate that someting wasn't being respected.

_Aside: I'd argue that it's not "really" a bug as that's a SUPER edge-case scenario brought on by a bad design._